src/server/frontend_wayland/text_input_v1.cpp

```diff
@@ -386,25 +394,30 @@ void TextInputV1::hide_input_panel()
 
 void TextInputV1::reset()
 {
-    pending_state.reset();
+    if(state == State::inactive)
+    {
+        mir::log_warning("Attempted to reset state while inactive");
+        return;
+    }
```

Comment:
Is this check worthwhile? It is likely to spam the log to no benefit.

If it is needed, we should probably be identifying the guilty client.

Reply:
It will be helpful to identify misbehaving clients in bug reports in the future. And from what I saw when debugging, `reset` isn't called on every frame. Rather, it's called on specific events. So it shouldn't be spamming logs unless a client is real naughty.

I tried identifying the client through `Session::name` but it comes up as empty. Maybe there's another way that I didn't see?

Reply:
Is there a bug to report?

                Should be called by an editor widget when the input state should be
                reset, for example after the text was changed outside of the normal
                input method flow.

There's nothing there that says a client is misbehaving by requesting `reset()` when inactive